scylla/src/transport/load_balancing/default.rs

```diff
@@ -170,8 +170,22 @@ or refrain from preferring datacenters (which may ban all other datacenters, if
         // some alive local node.
         // If there was no preferred datacenter specified, all nodes are treated as local.
         let nodes = self.preferred_node_set(cluster);
-        let picked = Self::pick_node(nodes, &self.pick_predicate);
-        if let Some(alive_local) = picked {
+
```

Comment:
The `token_with_strategy.is_some()` and `token_with_strategy_is_none()` paths look very similar. Perhaps we could unify them somehow? Not necessarily here, just food for thought for future improvements.